siliconcompiler/core.py

```diff
@@ -85,6 +85,10 @@ def __init__(self, design, loglevel=None):
             self.error("""SiliconCompiler must be run from a directory that exists.
 If you are sure that your working directory is valid, try running `cd $(pwd)`.""", fatal=True)
 
+        # Indicates that the chip object is marked as read only and should not
+        # allow modifications to schema
+        self._readonly = False
```

Comment:
Could we remove the attribute-based `_readonly` flag, and leave the `Chip` object mutability in place until we can land on a read-only Chip implementation (/decide if we want one)?

Reply:
Sounds reasonable. With the larger refactor making a ReadOnly schema would be easy